lib/scholar/linear_regression.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Scholar.LinearRegression do
```

Comment:
Do we want to namespace this inside `Scholar.Stats` or similar?

Reply:
For sklearn this is namespaced under `linear_model` with LogisticRegression and a couple of others. We can go that route. I think it depends how we want to structure various models / algorithms

Reply:
I think we have three options here:

1. Have a single module with a bunch of functions. All of those functions reuse the same struct that would have be like `Scholar.Linear{op: :linear_regression, data: ...}`

2. Have one module per operation, each with their own struct. Then I think we should namespace because the collection is too large

3. Have one module per operation with a shared struct

Looking at sklearn, the parameters of linear structs do not actually match. LinearRegression has different parameters than ElasticNet, so it makes me think 2 is the best option here.

However, I am wondering if we should go the protocol route. The current approach means that, for each model, we will have to define at least 3 modules: the model module, the impl for Scholar, and the impl for Nx.Container. Likely others in the future. So I am wondering if instead we should make `Scholar` a behaviour and merge the model module with the impl module. In any case, it can be done in a future PR.

TL;DR - yes, let's namespace it.